packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -1110,6 +1110,35 @@ export function authorizationServiceBuilder(
         )
       );
     },
+    async getProducerKeychainKeys({
+      producerKeychainId,
+      userIds,
+      organizationId,
+      logger,
+    }: {
+      producerKeychainId: ProducerKeychainId;
+      userIds: UserId[];
+      organizationId: TenantId;
+      logger: Logger;
+    }): Promise<Key[]> {
+      logger.info(
+        `Retrieving keys for producer keychain ${producerKeychainId}`
+      );
+      const producerKeychain = await retrieveProducerKeychain(
+        producerKeychainId,
+        readModelService
+      );
+      assertOrganizationIsProducerKeychainProducer(
+        organizationId,
+        producerKeychain.data
+      );
+      if (userIds.length > 0) {
+        return producerKeychain.data.keys.filter(
+          (k) => k.userId && userIds.includes(k.userId)
```

Comment:
I think we don't need to check if `userId` exists in the key, before using for the comparison. I see the usage is similar also in the `getClientKeys` endpoint, so the suggestion applies there too.